src/latexify/codegen/expression_codegen.py

```diff
@@ -240,6 +240,129 @@ def _generate_transpose(self, node: ast.Call) -> str | None:
+            return rf"\det \left( {self._generate_matrix(node)} \right)"
+
+        return None
+
+    def _generate_matrix_rank(self, node: ast.Call) -> str | None:
+        """Generates LaTeX for numpy.linalg.matrix_rank.
+        Args:
+            node: ast.Call node containing the appropriate method invocation.
+        Returns:
+            Generated LaTeX, or None if the node has unsupported syntax.
+        Raises:
+            LatexifyError: Unsupported argument type given.
+        """
+        name = ast_utils.extract_function_name_or_none(node)
+        assert name == "matrix_rank"
+
+        if len(node.args) != 1:
+            return None
+
+        func_arg = node.args[0]
+        if isinstance(func_arg, ast.Name):
+            return rf"\mathrm{{rank}} \left( \mathbf{{{func_arg.id}}} \right)"
+        elif isinstance(func_arg, ast.List):
+            return rf"\mathrm{{rank}} \left( {self._generate_matrix(node)} \right)"
+
+        return None
+
+    def _generate_matrix_power(self, node: ast.Call) -> str | None:
+        """Generates LaTeX for numpy.linalg.matrix_power.
+        Args:
+            node: ast.Call node containing the appropriate method invocation.
+        Returns:
+            Generated LaTeX, or None if the node has unsupported syntax.
+        Raises:
+            LatexifyError: Unsupported argument type given.
+        """
+        name = ast_utils.extract_function_name_or_none(node)
+        assert name == "matrix_power"
+
+        if len(node.args) != 2:
+            return None
+
+        func_arg = node.args[0]
+        power_arg = node.args[1]
+        if isinstance(power_arg, ast.Num):
+            if isinstance(func_arg, ast.Name):
+                return rf"\mathbf{{{func_arg.id}}}^{{{power_arg.n}}}"
+            elif isinstance(func_arg, ast.List):
+                matrix = self._generate_matrix(node)
+                if matrix is not None:
+                    return rf"{matrix}^{{{power_arg.n}}}"
+        return None
+
+    def _generate_qr_and_svd(self, node: ast.Call) -> str | None:
+        """Generates LaTeX for numpy.linalg.qr and numpy.linalg.svd.
+        Args:
+            node: ast.Call node containing the appropriate method invocation.
+        Returns:
+            Generated LaTeX, or None if the node has unsupported syntax.
+        Raises:
+            LatexifyError: Unsupported argument type given.
+        """
+        name = ast_utils.extract_function_name_or_none(node)
+        assert name == "QR" or name == "SVD"
+
+        if len(node.args) != 1:
+            return None
+
+        func_arg = node.args[0]
+        if isinstance(func_arg, ast.Name):
+            func_arg_str = rf"\mathbf{{{func_arg.id}}}"
+            return rf"\mathrm{{{name.upper()}}} \left( {func_arg_str} \right)"
+
+        elif isinstance(func_arg, ast.List):
+            matrix_str = self._generate_matrix(node)
+            return rf"\mathrm{{{name.upper()}}} \left( {matrix_str} \right)"
+        return None
```

Comment:
This function can be removed. The current implementation converts `QR(A)` to $\mathrm{QR}(A)$ and `SVD(A)` to $\mathrm{SVD}(A)$